pkg/utils/utils.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+// GetClusterDomainName returns cluster's domain name or error
+func GetClusterDomainName() (string, error) {
+	_, err := os.Stat("/etc/resolv.conf")
+	if err != nil {
+		return "", err
+	}
+	f, err := os.Open("/etc/resolv.conf")
```

Comment:
I think you could simplify this by substituting a `strings.NewReader("")` for `f`:

```golang
f, err := os.Open("/etc/resolv.conf")
if err != nil {
  fmt.Printf("Unable to open /etc/resolv.conf: %v", err)
  f = ioutil.NopCloser(strings.NewReader(""))
}
defer f.Close()
return getClusterDomainName(f)
```

Reply:
I am not sure I understand the logic here,  why would I need an empty string reader on a failure of reading /etc/resolv.conf. Appreciate a lot some clarification.